app/src/main/java/com/hedvig/app/feature/marketpicker/MarketPickerViewModel.kt

```diff
@@ -18,43 +18,31 @@ import com.hedvig.app.shouldOverrideFeatureFlags
 import com.hedvig.app.util.apollo.defaultLocale
 import com.hedvig.app.util.extensions.getLanguage
 import com.hedvig.app.util.extensions.getMarket
-import com.hedvig.app.util.extensions.getStoredBoolean
 import kotlinx.coroutines.launch
 
 abstract class MarketPickerViewModel(private val context: Context) : ViewModel() {
     protected val _data = MutableLiveData<PickerState>()
     val data: LiveData<PickerState> = _data
     abstract fun uploadLanguage()
 
+    fun submitLanguageAndReload() {
+        uploadLanguage()
```

Comment:
Let's investigate if this is the approach we want to take - I've got some additional context on why this thing was built this way 😬